### ISSUE TITLE
[bug] unable to use a type alias spec of an inlined haskell function in a refined data type spec

### ISSUE THREAD

**Issue by plredmond** [ucsd-progsys/liquidhaskell#1761]
**Problem:**
* Define and inline a predicate over a value.
* Create an alias as shorthand to express that the predicate is satisfied in a specification.
* Use the specification in a refined data type.
```haskell
{-@ inline oneFunPred @-}
oneFunPred :: Int -> Bool
oneFunPred x = x == 1

{-@ type OneTyAlias a = {v:a | oneFunPred v} @-}

{-@ data One = One { field :: OneTyAlias Int }  @-}
data One = One Int -- ERROR (ranjit said it ought to pass)
```
**Expect:**
The refined data type has a constructor with a specification requiring the predicate to be satisfied.
**Actual:**
Unbound symbol error on `oneFunPred` for the line with the data definition.
```
lib/Bug.hs:10:12: error:
    • Illegal type specification for `Bug.One`
    Bug.One :: Bug.field##Bug.One:{VV : GHC.Types.Int | Bug.oneFunPred VV} -> {VV : Bug.One | Bug.field VV == Bug.field##Bug.One
                                                                                              && Bug.field VV == Bug.field##Bug.One
                                                                                              && (is$Bug.One VV <=> true)}
    Sort Error in Refinement: {VV : int | Bug.oneFunPred VV}
    Unbound symbol Bug.oneFunPred --- perhaps you meant: Bug.field, Bug.One ?
    •
   |
10 | data One = One Int
   |            ^^^^^^^
```

---

**Workaround:**
Inline the type alias into the refined data type specification.
```haskell
{-@ inline oneFunPred @-}
oneFunPred :: Int -> Bool
oneFunPred x = x == 1

{-@ data One = One { field :: {v:Int | oneFunPred v}}  @-}
data One = One Int

-- This example passes (as expected)
```

**Version:**
Bug present in hackage releases
```
liquidhaskell 0.8.10.2
liquid-base 4.14.1.0
liquid-prelude 0.8.10.2
```
and also present in built-from-source
```
liquidhaskell@de4f003920d23a2683987c7f89db1afcf7df3b80 (Sept 16)
```

[Link to slack conv](https://liquidhaskell.slack.com/archives/C54QAL9RR/p1600451110042700)

**Comment by plredmond** [comment]
Tagging @adinapoli per @.ranjitjhala's comment [in slack](https://liquidhaskell.slack.com/archives/C54QAL9RR/p1600469137047800?thread_ts=1600451110.042700&cid=C54QAL9RR).

**Comment by adinapoli** [comment]
I think this is a different issue from the type-alias one I fixed as part of #1745. If we take a look at the error:

```
      T1761.hs: FAIL (4.90s)
        tests/test.hs:503:
        Wrong exit code; output:
          > 
          > tests/pos/T1761.hs:12:12: error:
          >     • Illegal type specification for `T1761.One`
          >     T1761.One :: T1761.field##T1761.One:{VV : GHC.Types.Int | T1761.oneFunPred VV} -> {VV : T1761.One | T1761.field VV == T1761.field##T1761.One}
          >     Sort Error in Refinement: {VV : int | T1761.oneFunPred VV}
          >     Unbound symbol T1761.oneFunPred --- perhaps you meant: T1761.One, T1761.field ?
          >     • 
          >    |
          > 12 | data One = One Int
          >    |            ^^^^^^^
```

We can see how the type alias is correctly expanded, but the `oneFunPred` can't be found. I smell another name resolution problem. I'll add a bunch of traces and investigate this.

**Comment by plredmond** [comment]
Thank you for taking a look.
